scripts/prepare_network.py

```diff
@@ -272,42 +282,73 @@ def set_line_nom_max(n):
     for o in opts:
         m = re.match(r"^\d+seg$", o, re.IGNORECASE)
         if m is not None:
-            n = apply_time_segmentation(n, m.group(0)[:-3])
+            solver_name = snakemake.config["solving"]["solver"]["name"]
+            n = apply_time_segmentation(n, m.group(0)[:-3], solver_name)
             break
 
     for o in opts:
```

Comment:
Maybe few comments here with the description of what the different options are may help the maintenance of the code: specifying the scope of the different options may help

Reply:
These options are documented in readthedocs: https://pypsa-meets-africa.readthedocs.io/en/latest/wildcards.html

Reply:
Mmm ok, though, I am a fan of short comments like # just to guide, they don't harm but ok!